test/clj/matcher_combinators/core_test.clj

```diff
@@ -373,6 +373,44 @@
                                     :in-any-order)
               ::result/weight 1})))
 
+(tabular
+  (fact "matching for absence in map"
+    (core/match (?matcher {:a (equals 42)
+                           :b absent})
+      {:a 42})
+    => (just {::result/type   :match
+              ::result/value  {:a 42}
+              ::result/weight 0})
+
+    (core/match (?matcher {:a (equals 42)
+                           :b absent})
+      {:a 42
+       :b 43})
+    => (just {::result/type   :mismatch
+              ::result/value  (just {:a 42
+                                     :b (just {:actual 43})})
```

Comment:
this error doesn't say it should be missing...

I'd like to see a print of a failure message on the terminal.

Reply:
The mismatch data-structures have pretty printers that turn these raw maps into output, so you don't really see the output in tests.

https://github.com/nubank/matcher-combinators/pull/66/commits/e6534d5898ff3583ed4b939190649da53c11c4bd is an example where you can tweak the values and rerun the tests to see the failure messages

for 
```clojure
(fact   {:a {:aa 11} :b 1} => (match {:a {:bb m/absent} :b m/absent}))
```
you get
```
FAIL map absent at (midje_test.clj:41)
Actual result did not agree with the checking function.
Actual result:
{:a {:aa 11} :b 1}
Checking function: (match {:a {:bb m/absent}, :b m/absent})
    The checker said this about the reason:
        [:mismatch {:a {:aa 11}, :b (unexpected 1)}]
```

and 

```clojure
(fact "absent used in wrong context"
  [1] =not=> (match [m/absent]))
```
you get 
```
FAIL absent used in wrong context at (midje_test.clj:44)
Actual result did not agree with the checking function.
Actual result:
[1]
Checking function: (match [m/absent])
    The checker said this about the reason:
        [:mismatch
 [(invalid-matcher-context
   "`absent` matcher should only be used as the value in a map")]]
```